specs/CustomDownload.md

```diff
@@ -0,0 +1,784 @@
+    out with an explanation either here or in the Examples
+-->
+
+
+# API Notes
+<!-- TEMPLATE
+    Option 1: Give a one or two line description of each API (type and member),
+        or at least the ones that aren't obvious from their name. These
+        descriptions are what show up in IntelliSense. For properties, specify
+        the default value of the property if it isn't the type's default (for
+        example an int-typed property that doesn't default to zero.)
+
+    Option 2: Put these descriptions in the below API Details section,
+        with a "///" comment above the member or type.
+-->
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+<!-- TEMPLATE
+    The exact API, in IDL format for our COM API and
+    in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/)
+    when possible, or in C# if starting with an API sketch for our .NET and WinRT API.
+
+    Include every new or modified type but use // ... to remove any methods,
+    properties, or events that are unchanged.
+
+    (GitHub's markdown syntax formatter does not (yet) know about MIDL3, so
+    use ```c# instead even when writing MIDL3.)
+
+    Example:
+
+    ```
+    /// Event args for the NewWindowRequested event. The event is fired when content
+    /// inside webview requested to open a new window (through window.open() and so on.)
+    [uuid(34acb11c-fc37-4418-9132-f9c21d1eafb9), object, pointer_default(unique)]
+    interface ICoreWebView2NewWindowRequestedEventArgs : IUnknown
+    {
+        // ...
+
+        /// Window features specified by the window.open call.
+        /// These features can be considered for positioning and sizing of
+        /// new webview windows.
+        [propget] HRESULT WindowFeatures([out, retval] ICoreWebView2WindowFeatures** windowFeatures);
+    }
+    ```
+
+    ```c# (but really MIDL3)
+    public class CoreWebView2NewWindowRequestedEventArgs
+    {
+        // ...
+
+	       public CoreWebView2WindowFeatures WindowFeatures { get; }
+    }
+    ```
+-->
+## Win32 C++
+```c#
+// Enums and structs
+
+typedef enum COREWEBVIEW2_DOWNLOAD_STATE {
+  /// The download is in progress.
+  COREWEBVIEW2_DOWNLOAD_STATE_IN_PROGRESS,
+  /// The connection with the file host was broken. The `InterruptReason` property
+  /// can be accessed from `ICoreWebView2DownloadStateChangedEventArgs`. See
+  /// `COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON` for descriptions of kinds of
+  /// interrupt reasons. Host can check whether an interrupted download can be
+  /// resumed with the `CanResume` property on the `ICoreWebView2Download`. Once
+  /// resumed, a download is in the `COREWEBVIEW2_DOWNLOAD_STATE_IN_PROGRESS` state.
+  COREWEBVIEW2_DOWNLOAD_STATE_INTERRUPTED,
+  /// The download completed successfully.
+  COREWEBVIEW2_DOWNLOAD_STATE_COMPLETED,
+} COREWEBVIEW2_DOWNLOAD_STATE;
+
+typedef enum COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON {
+  COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON_NONE,
+
+  /// File errors
```

Comment:
We don't have a way to do categories like this. It will just turn into documentation for the FILE_FAILED enum entry not a heading for all of the file enum entries.
```suggestion
```